dataactcore/read_config.py

```diff
@@ -58,12 +51,8 @@ def merge_config(existing_config, new_config):
             continue
     # File successfully loaded, mark as no longer first
     first_config = False
-
-    CONFIG_BROKER = merge_config(CONFIG_BROKER, CONFIG_ALL['broker']) if "broker" in CONFIG_ALL else CONFIG_BROKER
-    CONFIG_SERVICES = merge_config(CONFIG_SERVICES, CONFIG_ALL['services']) if "services" in CONFIG_ALL else CONFIG_SERVICES
-    CONFIG_DB = merge_config(CONFIG_DB, CONFIG_ALL['db']) if "db" in CONFIG_ALL else CONFIG_DB
-    CONFIG_LOGGING = merge_config(CONFIG_LOGGING, CONFIG_ALL['logging']) if "logging" in CONFIG_ALL else CONFIG_LOGGING
-    CONFIG_JOB_QUEUE = merge_config(CONFIG_JOB_QUEUE, CONFIG_ALL['job-queue']) if "job-queue" in CONFIG_ALL else CONFIG_JOB_QUEUE
+    for category_name in CONFIG_CATEGORIES:
+        locals()[CONFIG_CATEGORIES[category_name]].update(CONFIG_ALL.get(category_name, {}))
```

Comment:
Sorry all of the questions about `local()`. I feel like I must be missing something about what's happening here, but I'm not sure why it's needed.